### PR TITLE
Added defaultValue to ManualDecorators.

### PR DESCRIPTION
**Pull request by niegowski**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Feature: Added defaultValue to ManualDecorators. Closes ckeditor/ckeditor5#6031.

---

### CODE REVIEW

File: tests/linkcommand.js [hunk 1]
```diff
@@ -324,25 +332,33 @@ describe( 'LinkCommand', () => {
 
 				expect( getData( model ) ).to.equal( 'foo[<$text linkHref="url">url</$text>]bar' );
 			} );
+
+			it( 'should add additional attributes to link if those are falsy but decorator\'s defaultValue is set to true', () => {
```
Comment (Reinmar) [inline tests/linkcommand.js:336]: Why storing the `false` value in the model? Also, how do you actually remove the value from the model?

Comment (Reinmar) [inline tests/linkcommand.js:336]: Why storing the `false` value in the model? Also, how do you actually remove the value from the model?

Comment (Reinmar) [reply tests/linkcommand.js:336]: As I see this – the only thing that needs to change when `defalutValue` is `true` is the state of this button in the UI when you create a new link. Since it'll be on at that point, the command will be executed with `isSth:true` and then our job's done. The rest will work the way it works so far. Do I miss something?

**Comment by Reinmar** [comment]
Based on manual tests all work fine.

File: tests/linkcommand.js (hunk 1 above)
Comment (Reinmar) [reply tests/linkcommand.js:336]: I checked that I correctly read the code – it indeed puts the false value in the model.

Comment (niegowski) [reply tests/linkcommand.js:336]: As discussed on a call:
* `LinkFormView` is binding its `isOn` property to `value` of manual decorator,
* `value` of a manual decorator is updated every time `LinkCommand` is `refresh`ed,
* every time `value` is extracted from the model's attribute,
* we need to store `false` in the attribute to indicate that user already decided whether the manual decorator is supposed to be applied or not

**Comment by Reinmar** [comment]
After looking at this PR again, I still have doubts regarding keeping the false value in the model. Isn't it equal in meaning to the lack of the attribute?

The problem is that a decorator which default value is false has two modes in the model:

*   when its attribute is not present – it's off,
*   when its present – it's on.

The decorator which default value is true has 3 modes:

*   when its attribute is present and set to `true` – it's on,
*   when its attribute is present and set to `false` – it's off,
*   when its attribute is not present it's a mess – it's off up until opening the link balloon again where it's automatically set to "on" and then saved as such.

Unfortunately, from the model perspective this is such a big inconsistency, that I'd work on resolving it. For instance, a link might've been inserted by a tool which knows nothing about decorators. In this case, it will insert a link with just `linkHref` and no decorator set to `false`, which creates the situation I described in the 3rd point.

To sum up – there should always be only 2 modes. In case of decorators which default value is set to true:

*   when its attribute is present and set to `true` – it's on,
*   when its attribute is note present – it's off and we should assume that it was set to that value.

I know that how the UI is implemented makes this tricky, but there's definitely some option there.

File: src/linkediting.js [hunk 2]
```diff
@@ -187,6 +189,41 @@ export default class LinkEditing extends Plugin {
...
+
+					dispatcher.on( 'element', ( evt, data, conversionApi ) => {
+						const matcherResult = matcher.match( data.viewItem );
+
+						if ( !matcherResult ) {
+							return;
+						}
+
+						// We must check if all decorator values are set on view element.
+						const decoratorValue = Object.entries( decorator.attributes ).every( ( [ key, value ] ) => {
```
Comment (Reinmar) [inline src/linkediting.js:211]: I wanted to write that this comparison is not precise enough because you can have two decorators using the class attribute but I can see that this is not supported already. It's a pity because the engine treats each class separately (styles and classes are special types of attributes) and this could work. In fact, getData() works, but setData() breaks.

Comment (Reinmar) [reply src/linkediting.js:211]: I reported a ticket for that in ckeditor/ckeditor5#6571.

File: src/linkediting.js [hunk 3]
```diff
@@ -187,6 +189,41 @@ export default class LinkEditing extends Plugin {
...
+					} );
+
+					dispatcher.on( 'element', ( evt, data, conversionApi ) => {
+						const matcherResult = matcher.match( data.viewItem );
+
+						if ( !matcherResult ) {
+							return;
+						}
+
+						// We must check if all decorator values are set on view element.
```
Comment (Reinmar) [inline src/linkediting.js:210]:
```suggestion
						// We must check if all decorator values are set on view element.
						// The comparison is naive (doesn't handle granular classes or styles)
						// but link decorators don't support them anyway (#6571).
```
